src/Centreon/Infrastructure/RequestParameters/SqlRequestParametersTranslator.php

```diff
@@ -255,6 +255,14 @@ private function createQueryOnKeyValue(string $key, $valueOrArray): string
             || $searchOperator === RequestParameters::OPERATOR_NOT_LIKE
             || $searchOperator === RequestParameters::OPERATOR_REGEXP
         ) {
+            // We check the regex
+            if ($searchOperator === RequestParameters::OPERATOR_REGEXP) {
+                try {
+                    preg_match($mixedValue, '');
```

Comment:
the `try` and `catch` blocks are useless in this case because of the function `preg_match` don't throw an exception!

Reply:
Wrong, an error handler has been defined and all errors throw an exception.

Reply:
this is the response of the request with invalid regex
> {"code":"42000","message":"An error has occurred in a repository"}

the exception is the same as before
> SQLSTATE[42000]: Syntax error or access violation: 1139 Got error 'nothing to repeat at offset 0' from regexp

Reply:
if this worked there would be an exception with the message
> Bad regex format ...